modules/apps/object/object-web/src/main/resources/META-INF/resources/js/components/ModelBuilder/Diagram/Diagram.tsx

```diff
@@ -68,10 +71,20 @@ function DiagramBuilder() {
 		};
 	}>();
 
-	const store = useStore();
+	const edges = elements.filter((element) => isEdge(element)) as Edge<
+		ObjectRelationshipEdgeData
+	>[];
+
+	const nodes = elements.filter((element) => isNode(element)) as Node<
+		ObjectDefinitionNodeData
+	>[];
```

Comment:
Great observation @rebsilva! I believe we can remove the last `if` and use an `else` instead, since the elements array can only have these two options, so:

```
	const edges : Edge<ObjectRelationshipEdgeData>[] = [];

	const nodes : Node<ObjectDefinitionNodeData>[] = [];

	elements.forEach((element) => {
		if (isEdge(element)) {
			edges.push((element as Edge<ObjectRelationshipEdgeData>))
		}

		else {
			nodes.push(element as Node<ObjectDefinitionNodeData>)
		}
	});
```

Is that ok with you?

Reply:
perfect! i was in doubt if it could be another element instead of edge and node, so its fine to me!